openpype/modules/kitsu/plugins/publish/integrate_kitsu_note.py

```diff
@@ -27,9 +28,19 @@ def process(self, context):
 
             # Get note status, by default uses the task status for the note
             # if it is not specified in the configuration
-            note_status = kitsu_task["task_status"]["id"]
-
-            if self.set_status_note:
+            shortname = kitsu_task["task_status"]["short_name"].upper()
+            note_status = kitsu_task["task_status_id"]
+            if self.set_status_note and next(
```

Comment:
It could be me, but this is hard to read code. I highly recommend to create separate method which returns `True` or `False`.
```python
def match_status_conditins(self, short_name):
    for status_except in self.status_exceptions:
        is_same = shortname == status_except["short_name"].upper()
        if status_except["condition"] == "equal" and is_same:
            return True
        if status_except["condition"] == "not_equal" and not is_same:
            return True
    return False
```

Reply:
I'm changing the implementation based on Roy's suggestion on [PR 4551](https://github.com/ynput/OpenPype/pull/4551#discussion_r1136731042), so it should be more readable on next commit.

Reply:
Is this PR needed in that case?

Reply:
PR 4551 derives from this PR, but each introduce a different feature. Then if adding both features as a single PR is preferred, I don't mind.

Reply:
I think we could merge this one, now it's ready. https://github.com/ynput/OpenPype/pull/4551#discussion_r1136731042 will be rebased on it.